cpp/inc/bond/core/bond_fwd.h

```diff
@@ -89,4 +89,19 @@ class Serializer;
 template <typename Protocols = BuiltInProtocols, typename Writer>
 Serializer<Writer, Protocols> SerializeTo(Writer& output);
 
+// uses_static_parser
+template <typename Reader, typename Enable = void> struct
+uses_static_parser
+    : std::false_type {};
+
+template <typename Reader> struct
+uses_static_parser<Reader, typename boost::enable_if<
+    std::is_same<typename Reader::Parser, StaticParser<Reader&> > >::type>
+    : std::true_type {};
+
+template <typename Reader> struct
+uses_static_parser<Reader&>
+    : uses_static_parser<Reader> {};
```

Comment:
Why are we moving definitions to a header with forward declarations?

Reply:
Should I create a separate header for these? Or do you suggest to put them into some other specific header?

Reply:
I'm not clear on what issue is being fixed by this move, but this header is meant for forward declarations. Can you elaborate on the issue?

Reply:
omit_default.h uses `uses_static_parser`, but does not include protocol.h. And I think there might be a circular dependency between the two, if I just add protocol.h to omit_default.h. but I don't remember at the moment. Can double check later.

Reply:
It would be good if you could share a small repro of the issue.

Reply:
Moved declaration to `traits.h` and included that file in `omit_default.h`